crypto/ethsecp256k1/ethsecp256k1.go

```diff
@@ -28,54 +36,85 @@ const (
 // ----------------------------------------------------------------------------
 // secp256k1 Private Key
 
-var _ tmcrypto.PrivKey = PrivKey{}
-
-// PrivKey defines a type alias for an ecdsa.PrivateKey that implements
-// Tendermint's PrivateKey interface.
-type PrivKey []byte
+var (
+	_ cryptotypes.PrivKey  = &PrivKey{}
+	_ codec.AminoMarshaler = &PrivKey{}
+)
 
 // GenerateKey generates a new random private key. It returns an error upon
 // failure.
-func GenerateKey() (PrivKey, error) {
+func GenerateKey() (*PrivKey, error) {
 	priv, err := ethcrypto.GenerateKey()
 	if err != nil {
-		return PrivKey{}, err
+		return nil, err
 	}
 
-	return PrivKey(ethcrypto.FromECDSA(priv)), nil
+	return &PrivKey{
+		Key: ethcrypto.FromECDSA(priv),
+	}, nil
+}
+
+// Bytes returns the byte representation of the ECDSA Private Key.
+func (privKey *PrivKey) Bytes() []byte {
```

Comment:
I'm a bit confused here. I saw that both `pointer receiver` and `value receiver` are used on `PrivKey` (and for `PubKey` later), from what I understand, we use `pointer receiver` if we need to mutate the state inside the method, or it's too large to use `value receiver` because copying the whole thing would be too expensive. Just want to learn the reason behind, maybe I'm simply missing smth.

Reply:
>  we use pointer receiver if we need to mutate the state inside the method, or it's too large to use value receiver because copying the whole thing would be too expensive.

yup, will update